### PR TITLE
D3jsmap

### PR DESCRIPTION
**Pull request by lsat12357**

### CODE REVIEW

File: oregon/templates/home.html [hunk 1]
```diff
@@ -90,4 +42,10 @@
   </div><!-- end summary -->
   <div class="clear"><!-- --></div>
 {% endblock %}
- 
+{% block javascript %} {{ block.super }}
+    <script src="http://d3js.org/d3.v3.min.js"></script>
+    <script src="http://d3js.org/topojson.v1.min.js"></script>
```
Comment (jechols) [inline oregon/templates/home.html]: Let's get this code locally unless there's some specific reason not to.  When we switch to https, this will break without an https endpoint on their side.  If they switch to https before us, we'll have to fix the code to keep linking.  If they do a minor update that breaks anything we have, we'll have no idea what happened an no way to revert.

File: oregon/templates/site.html [hunk 2]
```diff
@@ -37,7 +37,12 @@
...
         margin-top: 10px;
     }
-
+    .place-label {
+    fill:#8C001A;
+    font-size:14;
+    font-weight: bold;
+    text-decoration:none !important;
+    }
     ul.newspaper_pages li.span4 {
```
Comment (jechols) [inline oregon/templates/site.html]: Fix the spacing here to be consistent with the rest of the styles

File: oregon/static/js/or2.json [hunk 3]
```diff
@@ -0,0 +1 @@
...[0,4],[-11,0],[0,-4]],[[4241,6199],[8,0],[0,3],[-8,0],[0,-3]],[[4017,4831],[6,0],[0,4],[-6,0],[0,-4]],[[5196,451],[10,0],[0,3],[-10,0],[0,-3]],[[3442,516],[14,0],[0,4],[-14,0],[0,-4]],[[2303,449],[9,0],[0,3],[-9,0],[0,-3]],[[2113,793],[9,0],[0,4],[-9,0],[0,-4]],[[1971,736],[12,0],[0,3],[-12,0],[0,-3]],[[1523,1021],[12,0],[0,3],[-12,0],[0,-3]],[[188,2619],[8,0],[0,3],[-8,0],[0,-3]],[[1511,2859],[10,0],[0,3],[-10,0],[0,-3]],[[413,3240],[10,0],[0,4],[-10,0],[0,-4]],[[926,3887],[11,0],[0,3],[-11,0],[0,-3]],[[1861,4220],[15,0],[0,3],[-15,0],[0,-3]],[[1906,4808],[11,0],[0,3],[-11,0],[0,-3]],[[1826,4822],[8,0],[0,3],[-8,0],[0,-3]],[[776,6159],[7,0],[0,4],[-7,0],[0,-4]],[[1889,6914],[7,0],[0,4],[-7,0],[0,-4]],[[2417,7888],[13,0],[0,3],[-13,0],[0,-3]],[[2332,8278],[8,0],[0,4],[-8,0],[0,-4]],[[890,8121],[11,0],[0,3],[-11,0],[0,-3]],[[2159,9064],[11,0],[0,3],[-11,0],[0,-3]],[[934,9840],[7,0],[0,3],[-7,0],[0,-3]],[[130,2093],[9,0],[0,4],[-9,0],[0,-4]],[[4289,7455],[8,0],[0,4],[-8,0],[0,-4]],[[1651,6697],[11,0],[0,4],[-11,0],[0,-4]],[[4070,6493],[15,0],[0,3],[-15,0],[0,-3]]],"transform":{"scale":[0.0008112574283090495,0.00042568292475525147],"translate":[-124.56676184799996,41.99954022137224]}}
```
Comment (jechols) [inline oregon/static/js/or2.json]: Can you run this through some kind of tidy-type thing for JSON?  If we need to change a single value, the diff is a lot easier if it's clearly changing just one line.

File: oregon/static/titlecities.txt [hunk 4]
```diff
@@ -0,0 +1,62 @@
+label,lat,lon,url
```
Comment (jechols) [inline oregon/static/titlecities.txt]: Is there any reason we couldn't put this data into `core_place`, and generate this file via a django-admin script?  (Perhaps something we add to the compile.sh script even) I'd prefer a single source for all city data, but I'm not sure if there was some reason for not choosing that table.

File: oregon/templates/site.html [hunk 5]
```diff
@@ -37,7 +37,12 @@
     #myCarousel {
         margin-top: 10px;
     }
-
+    .place-label {
+    fill:#8C001A;
+    font-size:14;
```
Comment (jechols) [inline oregon/templates/site.html]: I'm getting a CSS error here, and it's just ignoring the font size.  You have to specify the meaning of the number: `pt`, `px`, `em`, etc.

File: oregon/templates/site.html (hunk 2 above)
Comment (jechols) [inline oregon/templates/site.html]: And can we get `:focus` and `:hover` state added?  There are so many links it's hard to tell which one has mouse/keyboard focus.

File: oregon/static/js/or2.json (hunk 3 above)
Comment (lsat12357) [inline oregon/static/js/or2.json]: this is generated code; also should not have any reason to edit it.

File: oregon/static/titlecities.txt (hunk 4 above)
Comment (lsat12357) [inline oregon/static/titlecities.txt]: I think this works for now; can revisit how the file is generated at a later date.

File: oregon/static/js/or2.json (hunk 3 above)
Comment (jechols) [inline oregon/static/js/or2.json]: What generates it?  Is it something we can generate locally on deploy rather than storing it in the repository?

File: oregon/templates/site.html [hunk 6]
```diff
@@ -38,10 +38,16 @@
         margin-top: 10px;
     }
     .place-label {
-    fill:#8C001A;
-    font-size:14;
-    font-weight: bold;
-    text-decoration:none !important;
+        fill:#951916;
```
Comment (jechols) [inline oregon/templates/site.html]: Next time put these kinds of changes in their own commit - they aren't related to adding the JS files

File: oregon/templates/site.html [hunk 7]
```diff
@@ -47,7 +47,8 @@
         font-size:12px;
     }
     .place-label:focus{
-        fill:#630804;
+        fill:#F30;
+        text-decoration:underline !important;
```
Comment (jechols) [inline oregon/templates/site.html]: +1

File: oregon/static/js/or2.json (hunk 3 above)
Comment (lsat12357) [inline oregon/static/js/or2.json]: I used gdal, topojson, and a public-domain geo dataset. Since only the state-level data is being used  (no city data) from this file at this point, it should not be necessary to touch it (at least until the big one hits and the shape of Oregon changes).

Comment (jechols) [inline oregon/static/js/or2.json]: Okay.  I'd like you to document how we add a new city, at least (wiki is probably best...?).  Right now you're the only one who has any idea about this part of the system :)
